### DISCUSSION TITLE
Opting out from identity map

### DISCUSSION THREAD

**Discussion by kristijanhusak** [mikro-orm/mikro-orm#2755] [Q&A] [answered]
Is there any way to somehow opt out from identity map?
I like the idea, and I see how it's useful, but I had more situations where it caused me more problems than not.
Doctrine easily leverages it because PHP boots from start on each request, so the state is always fresh. JS doesn't work that way.

Implementing it in NestJS works great for regular requests, but if there is some functionality that doesn't go through it, it uses the global scope. Like [events](https://docs.nestjs.com/techniques/events#listening-to-events), [Login strategies](https://docs.nestjs.com/security/authentication#implementing-passport-local), and similar things.
I know I could set the whole thing from the controller level to be request scoped, but that can cause [performance issues](https://docs.nestjs.com/fundamentals/injection-scopes#performance)

Ideally, I would globally opt out from it, and use it only where I need it, something similar like a transaction.

**Comment by B4nan** [comment, marked as answer]
> JS doesn't work that way.

With AsyncLocalStorage, it actually can work the same. For anything that is not part of middlewares, you can use `@UseRequestContext()` decorator for the same.

You can always just fork, which creates separate context that will get garbage collected once you stop referencing it. The same is actually done if you use `disableIdentityMap` in the find options, just forks internally and clears afterward before returning.

I am not interested in global way of opting out of identity map.

**Reply by kristijanhusak** [reply]
`@UseRequestContext()` requires injecting the Mikro ORM instance into the class where you plan to use it. I don't want to create a dependency on ORM in my service classes.

I did some manual forking, but I didn't know about `disableIdentityMap`. I guess I'll just use that for these use cases.
Thanks.
